ui/components/app/confirmations/alerts/multiple-alert-modal/multiple-alert-modal.tsx

```diff
@@ -0,0 +1,161 @@
+  /** The function to be executed when the modal needs to be closed. */
+  onClose: () => void;
+  /** The function to be executed when the button in the alert modal is clicked. */
+  onFinalAcknowledgeClick: () => void;
+  /** The unique identifier of the entity that owns the alert. */
+  ownerId: string;
+};
+
+function PageNavigation({
+  alerts,
+  onBackButtonClick,
+  onNextButtonClick,
+  selectedIndex,
+}: {
+  alerts: Alert[];
+  onBackButtonClick: () => void;
+  onNextButtonClick: () => void;
+  selectedIndex: number;
+}) {
+  const t = useI18nContext();
+  const showPreviousButton = selectedIndex + 1 > 1;
+  const showNextButton = selectedIndex + 1 < alerts.length;
+
+  function PreviousButton() {
+    if (!showPreviousButton) {
+      return null;
+    }
+
+    return (
+      <ButtonIcon
+        iconName={IconName.ArrowLeft}
+        ariaLabel={t('back')}
+        size={ButtonIconSize.Sm}
+        onClick={onBackButtonClick}
+        className={'multiple-alert-modal__arrow-buttons'}
+        data-testid="alert-modal-back-button"
+      />
+    );
+  }
+
+  function NextButton() {
+    if (!showNextButton) {
+      return null;
+    }
+
+    return (
+      <ButtonIcon
+        iconName={IconName.ArrowRight}
+        ariaLabel={t('next')}
+        size={ButtonIconSize.Sm}
+        onClick={onNextButtonClick}
+        className={'multiple-alert-modal__arrow-buttons'}
+        data-testid="alert-modal-next-button"
+      />
+    );
+  }
+
+  function PageNumber() {
+    return (
+      <Text
+        variant={TextVariant.bodySm}
+        color={TextColor.textAlternative}
+        className={'multiple-alert-modal__text'}
+      >
+        {`${selectedIndex + 1} ${t('ofTextNofM')} ${alerts.length}`}
+      </Text>
+    );
+  }
+
+  return (
+    <Box display={Display.Flex}>
+      <PreviousButton />
+      <PageNumber />
+      <NextButton />
+    </Box>
+  );
+}
+
+export function MultipleAlertModal({
+  alertKey,
+  onClose,
+  onFinalAcknowledgeClick,
+  ownerId,
+}: MultipleAlertModalProps) {
+  const { alerts } = useAlerts(ownerId);
+
+  const [selectedIndex, setSelectedIndex] = useState(
+    alerts.findIndex((alert) => alert.key === alertKey),
+  );
+
+  const selectedAlert = useMemo(
+    () => alerts[selectedIndex],
+    [alerts, selectedIndex],
+  );
+  const onBackButtonClick = useCallback(() => {
+    setSelectedIndex((prevIndex) =>
+      prevIndex > 0 ? prevIndex - 1 : prevIndex,
+    );
+  }, []);
+
+  const onNextButtonClick = useCallback(() => {
+    setSelectedIndex((prevIndex) =>
+      prevIndex < alerts.length - 1 ? prevIndex + 1 : prevIndex,
+    );
+  }, [alerts.length]);
+
+  const handleAcknowledgeClick = useCallback(() => {
+    if (selectedIndex + 1 === alerts.length) {
+      onFinalAcknowledgeClick();
+      return;
+    }
+
+    onNextButtonClick();
+  }, [onFinalAcknowledgeClick, onNextButtonClick, selectedIndex, alerts]);
+
+  const renderPageNavigation = () => {
+    if (alerts.length <= 1) {
```

Comment:
Can this check be encapsulated within `PageNavigation` in which case we wouldn't need this `renderPageNavigation` function and could provide it inline below for greater readability?

Reply:
Done